File: app/src/main/java/com/cogu/spylook/DAO/AnotacionDAO.java
package com.cogu.spylook.DAO;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.cogu.spylook.model.entity.Anotacion;

import java.util.List;

@Dao
public interface AnotacionDAO {
    @Insert
    public void addAnotacion(Anotacion anotacion);

    @Query("Select * from anotaciones where id = :id")
    public Anotacion findAnotacionById(int id);

    @Query("Select * from anotaciones where contacto_id = :id")
    public Anotacion findAnotacionByContactoId(int id);

    @Query("Select * from anotaciones")
    public List<Anotacion> findAllAnotaciones();

}

File: app/src/main/java/com/cogu/spylook/DAO/AnotacionDAO.kt
package com.cogu.spylook.DAO

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Transaction
import androidx.room.Update
import com.cogu.spylook.model.entity.Anotacion
import com.cogu.spylook.model.relationships.ContactoAnotacion

@Dao
interface AnotacionDAO {

    @Insert
    suspend fun addAnotacion(anotacion: Anotacion)

    @Update
    fun updateAnotacion(anotacion: Anotacion)

    @Delete
    fun deleteAnotacion(anotacion: Anotacion)

    @Query("SELECT * FROM anotaciones WHERE id = :id")
    fun findAnotacionById(id: Int): Anotacion

    @Transaction
    @Query("SELECT * FROM contactos")
    fun getContactosWithAnotaciones():List<ContactoAnotacion>
}